src/AutoRest.CSharp/Common/Input/Inputs.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using AutoRest.CSharp.Input;
+using Azure.Core;
+
+#pragma warning disable SA1649
+namespace AutoRest.CSharp.Common.Input
+{
+    internal record InputNamespace(string Name, string Description, IReadOnlyList<InputClient> Clients);
+
+    internal record InputClient(string Name, string Description, IReadOnlyList<InputOperation> Operations);
+
+    internal record InputOperation(
+        string Name,
+        string Description,
+        string? Accessibility,
+        IReadOnlyList<InputOperationParameter> Parameters,
+        IReadOnlyList<OperationResponse> Responses,
+        RequestMethod HttpMethod,
+        BodyMediaType RequestBodyMediaType,
+        string Uri,
+        string Path,
+        string? ExternalDocsUrl,
+        IReadOnlyList<string>? RequestMediaTypes,
+        bool BufferResponse,
+        OperationLongRunning? LongRunning,
+        OperationPaging? Paging);
+
+    internal record InputOperationParameter(
+        string Name,
+        string NameInRequest,
+        string? Description,
+        InputType Type,
+        RequestLocation Location,
+        InputConstant? DefaultValue,
+        VirtualParameter? VirtualParameter,
+        InputOperationParameter? GroupedBy,
+        bool IsConstant,
+        bool IsRequired,
+        bool IsApiVersion,
+        bool IsResourceParameter,
+        bool IsContentType,
+        bool IsEndpoint,
+        bool IsFlattened,
+        bool IsInMethod,
```

Comment:
Where is the fix? Have you pushed?

Reply:
I think what Renhe means is there are so many boolean types here. He is expecting a flag eum?

Reply:
I don't want to have flag enum here:
- These values are meaningless without InputParameter, so enum would be useless by itself
- It is much easier to make an error while reading a value of the flag enum than when just check a boolean.
- Some of the values can be combined while some can't. I can put `IsEndpoint`, `IsApiVersion` and `IsContentType` into one non-flag enum cause they are mutually exclusive (parameter is either endpoing, API version or content type).

Reply:
I have an unrelated question: what does `IsResourceParameter` mean here? It is fetched from an extension `x-ms-resource-identifier` and it has the word `resource identifier` in it so that I have this question: what is this for? Does the resource identifier here mean the same thing as the resource identifier in arm?

Reply:
Sorry, but I don't know what resource identifier in arm is :) 
This extension marks parameters that, when not presented in operation, mean that operation doesn't belong to the resource client but rather to its parent client